internal/controller/controller.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import "sigs.k8s.io/controller-runtime/pkg/controller"
+
+// TODO: could make this more generic for any Runnable
+type noLeaderElectionController struct {
+	controller.Controller
+}
+
+func (*noLeaderElectionController) NeedLeaderElection() bool {
+	return false
```

Comment:
Does this just fulfill an interface or something?

Reply:
yep, see https://github.com/kubernetes-sigs/controller-runtime/blob/7e7bf8c1e1df991b957dfaa8b36d975740f309ab/pkg/manager/manager.go#L301-L306

could do something to ensure the type is assignable to a particular interface to do a compilation-time check, but also would need to check the return value of this method since it matters as well, which is a little unfortunate how they implemented it